cli/parser/parser.go

```diff
@@ -156,6 +156,100 @@ func (p *Parser) AddOptionDefinition(o *options.Definition) error {
 	return nil
 }
 
+func ParseArgs(args []string) (*parsed.OrderedArgs, error) {
+	p, err := GetParser()
+	if err != nil {
+		return nil, err
+	}
+	return p.ParseArgsForCommand(args, "startup")
+}
+
+func (p *Parser) ParseArgsForCommand(args []string, command string) (*parsed.OrderedArgs, error) {
+	parsedArgs := &parsed.OrderedArgs{}
+	next := args
+	for {
+		opts, argIndex, err := p.ParseOptions(next, command)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s options: %s", command, err)
+		}
+		parsedArgs.Args = append(parsedArgs.Args, arguments.FromConcrete(opts)...)
+		next = next[argIndex:]
+		if len(next) == 0 {
+			break
+		}
+		if next[0] == "--" {
+			if command == "startup" {
+				// Bazel does not recognize `--` until the command has been encountered.
+				return nil, fmt.Errorf("unknown startup option '--'.")
+			}
+			parsedArgs.Args = append(parsedArgs.Args, arguments.ToPositionalArguments(next)...)
+			break
+		}
+		if command == "startup" {
+			command = next[0]
+			if command == "" {
+				// bazel treats a blank command as a help command that halts both option and
+				// argument parsing and ignores all non-startup options in the rc file.
+				break
+			}
+			if _, ok := p.BazelCommands[command]; !ok {
+				return nil, fmt.Errorf("Command '%s' not found. Try 'bb help'", command)
+				/*
+					// This isn't a bazel command, but it might be a bb extension.
+					extensionParser := NewParser(nil)
+					extensionParser.BazelCommands[command] = struct{}{}
+					extensionArgs, err := extensionParser.ParseArgsForCommand(next[1:], command)
+					if err != nil {
+						return nil, err
+					}
+					parsedArgs.Args = append(parsedArgs.Args, &arguments.PositionalArgument{Value: next[0]})
+					parsedArgs.Args = append(parsedArgs.Args, extensionArgs.Args...)
+					return parsedArgs, nil
+				*/
```

Comment:
(?)

Reply:
lolllll, I was supposed to remove this. It would maybe be useful in the future, but is just a confounding factor for now and after talking with you the other day this isn't where it belongs anyway. Thanks!